### DISCUSSION TITLE
検索画面についての疑問と改良提案

### DISCUSSION THREAD

**Discussion by tsubolabo** [increments/qiita-discussions#361] [要望・アイデア・不具合]
『まだ "<", ">" に拘っているの？　シツコイわね！:eyes:』
『ごめんなさい🙇、別件です！』

　今回も、私の日々の検索方法だけしか考えていない、自己中心的な提案で申し訳ありません。他の方の反対があれば撤回します。

　検索結果の表示方法は「新着順」「関連順」「ストック数順」「いいね数順」から選べるようになっており、既定では「関連順」にチェックマークが入っています。ところで、この **「関連順」とはどんな意味なのでしょう？**。自分がフォローしているタグに関係ある順でもなさそうだし、いったい何なんでしょう？　使っているうちに分かるだろうと思っていましたが未だに分かりません。

　まず、最初の提案は、この**既定を「新着順」に変えて頂きたい**ということです。

　第２の提案は、**「検索オプション」欄の「created:>ひと月前の今日の日付」を「created:>一昨日の日付」に変えていただきたい**ということです。できれば、さらに、これを検索オプションの第１行目のアクセスしやすい位置に移動していただければなお結構です。

　ひと月前の設定のままでは、約7000件もの記事が該当してしまいます。100ページの上限にはとても収まりきらない量です。しかし、一昨日の日付にすれば300～400件で丁度良い量になります。

＝＝＝　私の現在の利用法　＝＝＝

　次に、私の検索の利用法を書いておきます。投稿を毎日チェックしているQiita依存症の私（難しいことは分からず、実際に読む記事はほんの一部です）が、これらの手間を減らしたいというのが提案動機です。極めてシンプルです。

手順：
+ ①　トップページの検索ワード欄に、適当な半角英字１文字を打ち込んで検索ページを表示する。
+ ②　検索オプションの「creared:」をクリックして、それを検索ページの検索ワード欄に転写する。
+ ③　転写された文字列の「月の数字」を先月から今月に書き換える。
+ ④　同じく、「日の数字」を今日から一昨日に書き換える。
+ ⑤　検索ボタンをクリックする。
+ ⑥　表示方法を「関連順」から「新着順」に変える。

これで、今日と昨日の投稿記事だけが、新しい順にソートされて選び出されます。

ここで反論が予想されます。

>現在の仕様のままでも、新着順に設定替えするだけで、同じ内容の検索結果が表示できるのではないか？

　確かにそうではありますが、「< 現ページ/全ページ >」表示の全ページの数字が大き過ぎると、終わりの見えない果てしなさがストレスになります。「あと何ページだ、頑張ろう！」の気持ちが湧きません。

**Comment by tanaka-asami415** [comment]
@tsubolabo 
Qiitaをご活用いただきありがとうございます。

> 既定を「新着順」に変えて頂きたい

恐れ入りますが、現時点では規定値を新着順に変える予定はありません。
すぐには難しいですが将来的に何かしらの方法を検討したいと思います🙇

関連順は詳細をお伝えすることはできませんが、記事の内容など弊社独自のアルゴリズムで表示されております。

> 「検索オプション」欄の「created:>ひと月前の今日の日付」を「created:>一昨日の日付」に変えていただきたい

オプションは、利用する際にユーザーさまが応用しやすい見本として、現在の記載になっておりますので
一昨日という具体的な日にちにすることは、運営判断としては難しいです🙇

tsubolaboさんのような検索機能活用を想定しておらず、ご要望に対応しきれず申し訳ないです。
取り急ぎの代案として、新着順をブックマークしていただければ、少しはお手間が軽減されるのではないかと思いますので、ご検討いただけると幸いです。

また、新しい投稿を見たい方のために、[タイムライン](https://qiita.com/timeline)という機能を提供しています。
自分の興味のあるタグやユーザー、Organizationをフォローして新着記事を閲覧する方法も併せてご活用ください。
